src/ploomber_engine/tracking/tracking.py

```diff
@@ -162,7 +162,9 @@ def _parse_cli_parameters(parameters):
 
 
 @telemetry.log_call("track-execution")
-def track_execution(filename, parameters, database="experiments.db", quiet=False):
+def track_execution(
+    filename, parameters=dict(), database="experiments.db", quiet=False
```

Comment:
`dict()` should not be used as default, as it leads to side-effects, better to do:

```suggestion
    filename, parameters=None, database="experiments.db", quiet=False
```

and then, at the top of the function:

```python
parameters = parameters or dict()
```

Reply:
Learned 👍 
https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python